src/Microsoft.AspNetCore.Mvc.DataAnnotations/Internal/DataAnnotationsMetadataProvider.cs

```diff
@@ -90,7 +95,17 @@ public void CreateDisplayMetadata(DisplayMetadataProviderContext context)
 
 
             var containerType = context.Key.ContainerType ?? context.Key.ModelType;
-            var localizer = _stringLocalizerFactory?.Create(containerType);
+            IStringLocalizer localizer;
+
+            if (_stringLocalizerFactory != null && _options?.Value?.DataAnnotationLocalizerProvider != null)
```

Comment:
Simplify to `_localizationOptions.DataAnnotationLocalizerProvider` because `Value` can't be `null` in real life. Note all other `DataAnnotationLocalizerProvider` consumers avoid `null`-propagation operators in their equivalent conditions.